src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1294,3 +1299,39 @@ def validate(self) -> None:
 
 
 yaml.add_representer(SNMPGatewaySpec, ServiceSpec.yaml_representer)
+class TracingSpec(ServiceSpec):
+    def __init__(self,
+                 service_type: str,
+                 es_nodes: Optional[str] = None,
+                 service_id: Optional[str] = None,
+                 config: Optional[Dict[str, str]] = None,
+                 networks: Optional[List[str]] = None,
+                 placement: Optional[PlacementSpec] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 port: Optional[int] = None,
+                 ):
+        assert service_type in ['elasticsearch', 'jaeger-agent', 'jaeger-collector', 'jaeger-query']
+
+        super(TracingSpec, self).__init__(
+            service_type, service_id,
+            placement=placement, unmanaged=unmanaged,
+            preview_only=preview_only, config=config,
+            networks=networks)
+        self.service_type = service_type
+        self.es_nodes = es_nodes
+        self.port = port    
+
+    def get_port_start(self) -> List[int]:
+        return [self.get_port()]
+
+    def get_port(self) -> int:
+        if self.port:
+            return self.port
+        else:
+            return {'elasticsearch':9200,
+                    'jaeger-agent': 6831,
+                    'jaeger-collector': 14250,
+                    'jaeger-query': 16686}[self.service_type]
```

Comment:
this is a copy of the monitoring spec, right?

Reply:
kind of,
we still need the `es_nodes` data member, so it will be passed to jaeger-collector/query